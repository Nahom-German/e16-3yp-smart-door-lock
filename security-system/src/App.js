
import './App.css';
import React, {Component} from 'react';
import Navbar from "./components/Navbar.js";
import firebase from './config/firebase.js';
import Login from "./Login.js"
import Home from "./Home.js"
import { BrowserRouter as Router, Link, NavLink, Redirect, Switch,Prompt} from 'react-router-dom';
import Route from 'react-router-dom/Route';
import SignIn from "./SignIn.js"
import ChangePW from "./ChangePW.js"
import AddData from "./AddData.js";
import ViewHistory from "./ViewHistory";
import Edit from "./Edit.js";
import ChangeNumber from "./ChangeNumber.js";
import SignUp from './SignUp.js';
import Notification from "./Notification.js";
import userhistory from "./UserHistory.js";
import AdminHistory from "./AdminHistory.js";


class App extends Component {
  constructor(props){
    super(props);
    this.state={
      user : {}

      
    }
  }
  render(){
    return (
      
        <Router>
          <div className="container">
             <Navbar/>
            
            <Switch>
            <Route path="/Signin" exact strict component={SignIn}/>
            <Route path ="/home" exact strict component ={Home}/>
            <Route path ="/login" exact strict component ={Login}/>
            <Route path ="/ChangePW" exact strict component ={ChangePW}/>
            <Route path ="/AddData" exact strict component ={AddData}/>
            <Route path ="/ViewHistory" exact strict component ={ViewHistory}/>
            <Route path ="/Edit" exact strict component ={Edit}/>
            <Route path ="/ChangeNumber" exact strict component ={ChangeNumber}/>
            <Route path ="/Notification" exact strict component ={Notification}/>
            <Route path ="/Signup" exact strict component ={SignUp}/>
            <Route path ="/userhistory" exact strict component ={userhistory}/>
            <Route path ="/AdminHistory" exact strict component ={AdminHistory}/>
            <Route exact path="/" component={Home} />
            
            </Switch>
          </div>
        </Router>
        
      
    );
  }
}

export default App;
