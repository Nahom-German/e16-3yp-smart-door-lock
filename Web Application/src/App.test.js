import { render, screen } from '@testing-library/react';
import App from './App';
import Enzyme ,{shallow,mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
/*
test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
*/
describe('App',()=>{
  it("app renders correctly. matches the snapshot", () => {
    const tree= renderer.create(
      <App />
    ).toJSON();
  
    expect(tree).toMatchSnapshot();
  });
  it("app routers renders correctly. matches the snapshot", () => {
    const wrapper= mount(
      <App />
    );
    let content =wrapper.find('container');
    
    expect(toJson(content)).toMatchSnapshot();
  });
  
  });
  