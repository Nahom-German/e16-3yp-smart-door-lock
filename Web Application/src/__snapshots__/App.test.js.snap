// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App app renders correctly. matches the snapshot 1`] = `
<div
  className="container"
>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      Administrator
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarColor01"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/home"
            onClick={[Function]}
          >
            Security Management 
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div>
    <div>
      <img
        height="500"
        src="/images/image1.jpg"
        width="100%"
      />
    </div>
    <button
      class="btn btn-primary"
      id="logoutbtn"
      onClick={[Function]}
      type="submit"
    >
      Log out
    </button>
  </div>
</div>
`;

exports[`App app routers renders correctly. matches the snapshot 1`] = `null`;
